controllers/hetznerbaremetalmachine_controller_test.go

```diff
@@ -825,4 +825,154 @@ var _ = Describe("HetznerBareMetalMachineReconciler", func() {
 			})
 		})
 	})
+
+	Context("Test EnqueueRequestsFromMapFunc BareMetalHostToBareMetalMachines", func() {
```

Comment:
Why do you test this with envtest and with relying on the events? You can easily test this in the same way and via a stable way through unit tests that use the Kubernetes API.. Just call BareMetalHostToBareMetalMachines directly!

Reply:
I could write a test for BareMetalHostToBareMetalMachines, but this would not show that this function would resolve the issue.

Test Driven Development means: First write a test which shows that the current code is broken or is missing a feature. Then write the code to make the test pass. I do not do that always. But in this case, I would really want to see that my fix is fixing something.

Reply:
@guettli I'm totally up for test-driven development. But you can do that as well with a normal unit test that would show you in a more consistent way that it works.

Reply:
How to write a unit-test which shows that the hbmm waits "forever" and does not detect when a hbmh gets free?

Reply:
@janiskemper the test was re-written to not use sleep(). Additionally, a unit-test was added. Please review again.